compiler/damlc/tests/daml-test-files/InterfaceNonConstructorViewtype.daml

```diff
@@ -0,0 +1,10 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- @SINCE-LF-FEATURE DAML_INTERFACE
+-- @ERROR expected monomorphic record type in view type, but found a non-type-constructor:
```

Comment:
this one is a bit confusing because `Int` _is_ a (nullary) type constructor, see https://github.com/digital-asset/daml/pull/14932#discussion_r962826327

Reply:
improved messages now report this as a built-in